### PR TITLE
commit with Entry point

### PR DESCRIPTION
**Pull request by Masembe0757**
This is my first pull request

### CODE REVIEW

**Review by fredmugerwa — CHANGES_REQUESTED** [review]
@Masembe0757 Kindly remove, the .vscode and bin folders in your commit. I sent a documentation on how to use the .ignore files. 
Follow the scope of work assigned to you

**Comment by Masembe0757** [comment]
Made some changes

File: User_Reg_model_entryPoint/src/models/UserRegModel.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import java.lang.reflect.Array;
+import java.util.*;
+
+//This class provides a constructor that provides attributes expected of a user 
+class User{
+    String username;
+    String firstname;
+    String lastname;
+    Date dob;
+
```
Comment (fredmugerwa) [inline User_Reg_model_entryPoint/src/models/UserRegModel.java]: - Make fields private to enforce encapsulation.
- toString(), equals(), and hashCode() Methods: Override these methods for better object representation and comparison.
- Use better variable naming, foinstance dateOfBirth is more understandable to dob

**Review by fredmugerwa — CHANGES_REQUESTED** [review]
I am not seeing any reason of you having two classes in this file. Consider using only the User class

**Comment by Masembe0757** [comment]
made the changes

File: User_Reg_model_entryPoint/src/models/UserRegModel.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    public String toString() {
+        return "String : " + Optional.ofNullable(this.userName).orElse("Unknown") ;
+
+        }
+    
+    public static void main(String[] args) {
+        
+        
+    }
+
```
Comment (fredmugerwa) [inline User_Reg_model_entryPoint/src/models/UserRegModel.java]: - main method is not necessary in this UserRegModel class
- You must override hashCode() in every class that overrides equals(). Failure to do so will result in a violation of the general contract for Object.hashCode(), which will prevent your class from functioning properly in conjunction with all hash-based collections, including HashMap, HashSet, and Hashtable.
- Try to follow the package naming format, User_Reg_model_entrypoint seems off

**Review by fredmugerwa — CHANGES_REQUESTED** [review]
Thank you for the changes done, improve the areas noted below so that I approve.

File: User_Reg_model_entryPoint/src/models/UserRegModel.java (hunk 2 above)
Comment (Masembe0757) [reply User_Reg_model_entryPoint/src/models/UserRegModel.java]: Ohhh  thank you lemme fix it, my bad

**Review by fredmugerwa — APPROVED** [review]
Though look into branch naming conventions, work_to_review doesn't bring out meaning of the branch content.

**Pull request merged**